sandbox/grist/imports/import_csv.py

```diff
@@ -90,108 +90,117 @@ def parse_file(file_path, parse_options=None):
   """
   parse_options = parse_options or {}
 
-  with open(file_path, "rb") as f:
+  with codecs.open(file_path, "rb") as f:
+    sample = f.read(100000)
+    encoding = chardet.detect(sample)['encoding'] or "utf8"
+    # In addition, always prefer UTF8 over ASCII.
+    if encoding == 'ascii':
+      encoding = 'utf8'
+  log.info("Using encoding %s" % encoding)
+
+  with codecs.open(file_path, mode="r", encoding=encoding) as f:
     parsing_options, export_list = _parse_open_file(f, parse_options=parse_options)
     return parsing_options, export_list
 
 
+def _guess_dialect(file_obj):
+  try:
+    # Restrict allowed delimiters to prevent guessing other char than this list.
+    dialect = csv.Sniffer().sniff(file_obj.read(100000), delimiters=['\t', ',', ';', '|'])
+    log.info("Guessed dialect %s" % dict(dialect.__dict__))
+  except csv.Error:
+    log.info("Cannot guess dialect using Excel as fallback.")
+    return csv.excel
+  else:
+    return dialect
```

Comment:
`else` isn't really helping here, I think it'd be better to leave it in the `try` block.